dotnet/src/AutoGen.Ollama/Agent/OllamaAgent.cs

```diff
@@ -69,22 +70,28 @@ public class OllamaAgent : IStreamingAgent
             while (!reader.EndOfStream && !cancellationToken.IsCancellationRequested)
             {
                 string? line = await reader.ReadLineAsync();
-                if (string.IsNullOrWhiteSpace(line)) continue;
+                if (string.IsNullOrWhiteSpace(line))
+                {
+                    continue;
+                }
 
                 ChatResponseUpdate? update = JsonSerializer.Deserialize<ChatResponseUpdate>(line);
-                if (update != null)
+                if (update is { Done: false })
                 {
                     yield return new MessageEnvelope<ChatResponseUpdate>(update, from: Name);
                 }
+                else
+                {
+                    var finalUpdate = JsonSerializer.Deserialize<ChatResponse>(line) ?? throw new Exception("Failed to deserialize response");
```

Comment:
Should we use the async overload?

Reply:
Not here, the async overload works with stream, not string

Reply:
You are right :)